crates/node/service/src/actors/sequencer/actor.rs

```diff
@@ -0,0 +1,232 @@
+    async fn start_build(&mut self) -> Result<(), <Self as NodeActor>::Error> {
+        // If there is currently a block building job in-progress, do not start a new one.
+        if self.latest_payload_rx.is_some() {
+            return Ok(());
+        }
+
+        let unsafe_head = self.unsafe_head.borrow();
+        let l1_origin = self.l1_origin_selector.next_l1_origin(&unsafe_head).await.unwrap();
+
+        // TODO(clabby): Check for consistent L1 origin
+
+        info!(
+            target: "sequencer",
+            parent_num = unsafe_head.block_info.number,
+            l1_origin_num = l1_origin.number,
+            "Started sequencing new block"
+        );
+
+        // Build the payload attributes for the next block.
+        //
+        // TODO: Temp error handling
+        let mut attributes =
+            self.builder.prepare_payload_attributes(*unsafe_head, l1_origin.id()).await.unwrap();
+
+        // If the next L2 block is beyond the sequencer drift threshold, we must produce an empty
+        // block.
+        attributes.no_tx_pool = (attributes.payload_attributes.timestamp >
+            l1_origin.timestamp + self.cfg.max_sequencer_drift(l1_origin.timestamp))
+        .then_some(true);
+
+        // Do not include transactions in the first Ecotone block.
+        if self.cfg.is_first_ecotone_block(attributes.payload_attributes.timestamp) {
+            info!(target: "sequencer", "Sequencing ecotone upgrade block");
+            attributes.no_tx_pool = Some(true);
+        }
+
+        // Do not include transactions in the first Fjord block.
+        if self.cfg.is_first_fjord_block(attributes.payload_attributes.timestamp) {
+            info!(target: "sequencer", "Sequencing fjord upgrade block");
+            attributes.no_tx_pool = Some(true);
+        }
+
+        // Do not include transactions in the first Granite block.
+        if self.cfg.is_first_granite_block(attributes.payload_attributes.timestamp) {
+            info!(target: "sequencer", "Sequencing granite upgrade block");
+            attributes.no_tx_pool = Some(true);
+        }
+
+        // Do not include transactions in the first Holocene block.
+        if self.cfg.is_first_holocene_block(attributes.payload_attributes.timestamp) {
+            info!(target: "sequencer", "Sequencing holocene upgrade block");
+            attributes.no_tx_pool = Some(true);
+        }
+
+        // Do not include transactions in the first Isthmus block.
+        if self.cfg.is_first_isthmus_block(attributes.payload_attributes.timestamp) {
+            info!(target: "sequencer", "Sequencing isthmus upgrade block");
+            attributes.no_tx_pool = Some(true);
+        }
+
+        // Do not include transactions in the first Interop block.
+        if self.cfg.is_first_interop_block(attributes.payload_attributes.timestamp) {
+            info!(target: "sequencer", "Sequencing interop upgrade block");
+            attributes.no_tx_pool = Some(true);
+        }
```

Comment:
Is there a way we can make this more future-proof so we don't have to come back here and update?

Perhaps we could create a `FirstBlock` variant that is returned by a new `first_block(timestamp)` method on the rollup config. Then we could match against the variants like `FirstBlock::Interop`. Then, when a new hardfork is added to the `HardforkConfig`, the `FirstBlock` must be updated or a compile error occurs which will require this to be updated due to the `match`, so it isn't silently forgotten. Something like that.